### ISSUE TITLE
`FAST002` rule produces false positives and incorrect suggestions

### ISSUE THREAD

**Issue by MarkusSintonen** [astral-sh/ruff#15043]
Ruff (0.8.3) `FAST002` produces false positives and suggests incorrect `Annotated` format with FastAPI params having default values. Following example can not be converted into the `Annotated` format as FastAPI does not support this format with params having default values. Also `--unsafe-fixes` fixes it to the format the FastAPI wont accept.

Ruff FAST002 violation:
```python
def test_fast002_bug():
    from fastapi import FastAPI, Query
    app = FastAPI()

    @app.get("/test")
    def handler(echo: str = Query("")):  # FAST002 FastAPI dependency without `Annotated`
        return echo
```

But that is not compatible with FastAPI. It fails deeply with an AssertionError:
```python
def test_fast002_bug():
    from typing import Annotated
    from fastapi import FastAPI, Query
    app = FastAPI()

    # AssertionError: `Query` default value cannot be set in `Annotated` for 'echo'. Set the default value with `=` instead.
    @app.get("/test")
    def handler(echo: Annotated[str, Query("")]):
        return echo
```

Used FastAPI 0.115.5.

The rule is pretty nice otherwise so thanks for it! Hopefully we get it fixed.
